packages/modal/src/Modal/components/Root/Root.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { HTMLAttributes, useRef } from "react"
+import { createPortal } from "react-dom"
+import classnames from "classnames"
+import { motion, AnimatePresence } from "framer-motion"
+import { FocusOn } from "react-focus-on"
+import { Backdrop } from "../Backdrop"
+import { CloseButton } from "../CloseButton"
+import styles from "./Root.scss"
+
+const dropIn = {
+  hidden: {
+    y: "-100vh",
+    opacity: 0,
+  },
+  visible: {
+    y: "0",
+    opacity: 1,
+    transition: {
+      duration: 0.1,
+      type: "spring",
+      damping: 25,
+      stiffness: 500,
+    },
+  },
+  exit: {
+    y: "100vh",
+    opacity: 0,
+  },
+}
+
+export interface RootProps extends HTMLAttributes<HTMLDivElement> {
+  isOpen: boolean
+  accessibleLabelId: string
+  children: React.ReactNode
+  onDismiss: () => void
+  dismissOnBackdropClick?: boolean
+  dismissOnEscPress?: boolean
+  hideCloseButton?: boolean
```

Comment:
Maybe instead of 3 separate props, we could do a single one that takes an array? Especially if they all turn false if provided

eg.
```
disableDismiss={[“dismissOnBackDropClick”, "dismissOnEscPress"]}
```

Reply:
I don't mind that. Maybe we can slim that down a little bit to:
`disableDismiss={['backdrop', 'closeButton']}`

Following the discussion we had in the show and tell, we can leave out the esc press one. I don't think we've seen any cases of people wanting to remove it in the years we've had our current modal, and it would go against accessibility guidelines.

It's worth calling out that the close button one is a bit different in that it determines whether a UI element will show/hide, compared to the backdrop one which doesn't change anything visually, only the behaviour. Do we care that we're tying together two things that aren't consistent in what they do?

Reply:
Hmm yeah tricky one, what was the scenario for removing backdrop click? Is it better to enforce a particular pattern rather than give the option?

Reply:
What would the type of `disableDismiss` be? .....

``` 
type => disableDismiss: [backdrop: boolean, closeButton: boolean]

use => onClickOutside={disableDismiss[0] ? onDismiss : undefined}

story => disableDismiss={[true, true]}

```

Reply:
@gyfchong

Reply:
Oh that's true, a bit complex in real usage huh. Seems less necessary if the only options are to hide the close button and disable the backdrop; so feel free to ignore my suggestion 😄